README.md

```diff
@@ -540,6 +540,36 @@ function add(x, y) {
 }
 ```
 
+## Function statics and object callable properties
+
+### Flow
+
+You can use objects with callable properties as function with statics. This is useful when annotating memoized functions.
+
+```js
+type MemoizedFactorialType = {
+  cache: {
+    [number]: number,
+  },
+  [[call]](number): number,  // callable property
```

Comment:
I think `[[call]]: (number) => number` is a more commonly used syntax. Flow also supports `{(number) => number}` Maybe add examples for all syntaxes?

Reply:
I suppose you are referring to the `$call` property syntax?

```js
type MemoizedFactorialType = {
  cache: {
    [number]: number,
  },
  (number) => number, // is a syntax error
}
```

[Try Flow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAodAXAngBwKZgCyeAtnAJYBeeAJgGICGAxhnAE7kMwAquBAvGADeqMGCbMAFngBcw0WLABtAHYBXEgCM8bALpz1WnQBoFAX1NiAFIe1sAlGH4A+MLZNhgwMOQDOYBjBfLBUMBgAPMB02dgUAEgkYGAMNOydXdzZjT281X1owVjB4OABrYvJSggxJP1QzIA)

The `$call` property syntax is deprecated and removed therefore I think we should leave that out.

Reply:
> I suppose you are referring to the `$call` property syntax?
> 
> ```js
> type MemoizedFactorialType = {
>   cache: {
>     [number]: number,
>   },
>   (number) => number, // is a syntax error
> }
> ```
> 
> [Try Flow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAodAXAngBwKZgCyeAtnAJYBeeAJgGICGAxhnAE7kMwAquBAvGADeqMGCbMAFngBcw0WLABtAHYBXEgCM8bALpz1WnQBoFAX1NiAFIe1sAlGH4A+MLZNhgwMOQDOYBjBfLBUMBgAPMB02dgUAEgkYGAMNOydXdzZjT281X1owVjB4OABrYvJSggxJP1QzIA)
> 
> The `$call` property syntax is deprecated and removed therefore I think we should leave that out.

Sorry, I should've been more clear. What I meant was include all the "syntaxes":

 1. Your original example: `{ [[call]](number): number }` (explicit call property)
 2. The more common format of the above: `{ [[call]]: (number) => number }` (explicit call property)
 3. Implicit call property: `{ (number): number }` (implicit call property).

The reason why I suggested the second example is because I think it's more common in libdefs (maybe I am wrong), too. Also, I think it might be the easiest to read/notice from the three, but maybe that's just my bias.

Reply:
I see. Sorry about misreading the last comment. Will work on integrating them.